crates/auth/src/providers/github.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: Apache-2.0
+use crate::{error::Error, session::Session};
+
+use http::header::USER_AGENT;
+use serde::Deserialize;
+
+// TODO: move any remaining github specific functionality here
+#[derive(Debug, Deserialize)]
+pub struct AuthRequest {
+    pub code: String,
+    pub state: String,
+}
+
+#[derive(Copy, Clone)]
+pub struct GitHub;
+
+pub const PROVIDER: GitHub = GitHub;
+
+impl GitHub {
+    /// Validate if a session should be considered active.
+    pub async fn validate(&self, session: &Session) -> Result<String, Error> {
```

Comment:
This is only ever used in tests? It it expected that the application would call `Github::validate` on a session?

Shouldn't there be a `Session` trait with a `validate` method implemented by e.g. `Provider` enum?

Reply:
:+1: 
I think I can change how this is used to make it private. It really shouldn't be public because an invalid session is never going to be useful.